pkgs/development/libraries/mozc/default.nix

```diff
@@ -0,0 +1,234 @@
+      owner = "utuhiro78";
+      repo = "mozcdic-ut-neologd";
+      name = repo;
+      rev = "90e59c7707a5fe250c992c10c6ceb08a7ce7e652";
+      hash = "sha256-zY7K/J4OzBTQHrj8sF4s8xPqakoWHHMxWrvnvHT6oxE=";
+    })
+    (fetchFromGitHub rec {
+      owner = "utuhiro78";
+      repo = "mozcdic-ut-personal-names";
+      name = repo;
+      rev = "0e51584283f759cb6bfac1f19eda92c2c524eed4";
+      hash = "sha256-8cJmSp7rGxSHy+TElFpkXmNvMxYLw2yiAYJJb2AUyyE=";
+    })
+    (fetchFromGitHub rec {
+      owner = "utuhiro78";
+      repo = "mozcdic-ut-place-names";
+      name = repo;
+      rev = "478004504981e094af407f4fbd64cd65cf2d85dd";
+      hash = "sha256-3BwK+RuZMl5+femxQXNhDF/NE3qfX6u/+imqL2Uqd8I=";
+    })
+    (fetchFromGitHub rec {
+      owner = "utuhiro78";
+      repo = "mozcdic-ut-skk-jisyo";
+      name = repo;
+      rev = "43518e6ea033681580a515281668c85eb74a5b14";
+      hash = "sha256-05T//ulsS5HvOKPdOEG87/Yp8GgzOB2X3wG8Sds3uUU=";
+    })
+    (fetchFromGitHub rec {
+      owner = "utuhiro78";
+      repo = "mozcdic-ut-sudachidict";
+      name = repo;
+      rev = "1b5ad7cb51325fc9b42732887fede29d6a02bb07";
+      hash = "sha256-E4kq3YnxdYRFlv4KLSvf4NRoPhWXL8c89JIvqsjuy3o=";
+    })
+    (fetchurl rec {
+      name = "jawiki";
+      url = "https://dumps.wikimedia.org/${name}/20230820/${name}-20230820-all-titles-in-ns0.gz";
+      recursiveHash = true;
+      hash = "sha256-pOK4ThhOaup4t896Wx9jfsKSmWlQYn4Yy4pTiiwAFIs=";
+      downloadToTemp = true;
+      postFetch = ''
+        mkdir -p $out
+        mv $downloadedFile $out/${name}.gz
+      '';
+    })
+    (fetchFromGitHub rec {
+      owner = "musjj";
+      repo = "jp-zip-codes";
+      name = repo;
+      rev = "be9cd2ac888a34eba30a425c43a7e54f5187a649";
+      hash = "sha256-02CE/DJnqNXUF0OH3ptq2sb8UbtqKjLKx7Hf0izq+X4=";
+    })
+  ];
+
+  sourceRoot = ".";
+
+  nativeBuildInputs = [
+    glibcLocales
+    pkg-config
+    python3
+    ruby
+    wrapQtAppsHook
+  ];
+
+  buildInputs = [ qtbase ];
+
+  preBuild = ''
+    cd mozc/src
+  '';
+
+  LOCALE_ARCHIVE = "${glibcLocales}/lib/locale/locale-archive";
+  LC_ALL = "en_US.UTF-8";
+
+  inherit bazel;
+  removeRulesCC = false;
+  dontAddBazelOpts = true;
+
+  bazelFlags = [
+    "--config"
+    "oss_linux"
+    "--compilation_mode"
+    "opt"
+  ];
+
+  bazelTargets = [
+    "server:mozc_server"
+    "gui/tool:mozc_tool"
+  ];
+
+  fetchAttrs = {
+    postPatch = ''
+      substituteInPlace mozc/src/WORKSPACE.bazel \
+        --replace \
+          'url = "https://www.post.japanpost.jp/zipcode/dl/kogaki/zip/ken_all.zip"' \
+          "url = \"file://$PWD/jp-zip-codes/ken_all.zip\"" \
+        --replace \
+          'url = "https://www.post.japanpost.jp/zipcode/dl/jigyosyo/zip/jigyosyo.zip"' \
+          "url = \"file://$PWD/jp-zip-codes/jigyosyo.zip\""
+    '';
+
+    preInstall = ''
+      rm -rf $bazelOut/external/qt_linux
+    '';
+
+    sha256 = "sha256-TNeU39pmUXBn0NC5Zwu9p4I86LhT432whrhM63UPQ7U=";
```

Comment:
Better check if the hash is platform independent.

Reply:
The hash is now platform-independent because `qt_linux` is no longer vendored by bazel. The CI should error out otherwise, because of the `aarch64-linux` builder.